controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -92,14 +121,241 @@ func (r *StorageClassClaimReconciler) Reconcile(ctx context.Context, request rec
+	r.log.Info("Running StorageClassClaim controller in Consumer Mode")
+
+	providerClient, err := providerclient.NewProviderClient(
+		r.ctx,
+		r.storageCluster.Spec.ExternalStorage.StorageProviderEndpoint,
+		10*time.Second,
+	)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// Close client-side connections.
+	defer providerClient.Close()
+
+	if r.storageClassClaim.GetDeletionTimestamp().IsZero() {
+
+		// Check if finalizers are present, if not, add them.
+		if !contains(r.storageClassClaim.GetFinalizers(), StorageClassClaimFinalizer) {
+			storageClassClaimRef := klog.KRef(r.storageClassClaim.Name, r.storageClassClaim.Namespace)
+			r.log.Info("Finalizer not found for StorageClassClaim. Adding finalizer.", "StorageClassClaim", storageClassClaimRef)
+			r.storageClassClaim.SetFinalizers(append(r.storageClassClaim.GetFinalizers(), StorageClassClaimFinalizer))
+			if err := r.update(r.storageClassClaim); err != nil {
+				return reconcile.Result{}, fmt.Errorf("failed to update StorageClassClaim [%v] with finalizer: %s", storageClassClaimRef, err)
+			}
+		}
+
+		// Creation phase.
+		if r.storageClassClaim.Status.Phase == "" {
+			r.storageClassClaim.Status.Phase = ocsv1alpha1.StorageClassClaimCreating
+
+			// storageClassClaimStorageType is the storage type of the StorageClassClaim
+			var storageClassClaimStorageType providerclient.StorageType
+			switch r.storageClassClaim.Spec.Type {
+			case "blockpool":
+				storageClassClaimStorageType = FulfillstorageclassclaimrequestBlockpool
+			case "sharedfilesystem":
+				storageClassClaimStorageType = FulfillstorageclassclaimrequestSharedfilesystem
+			}
+
+			// Call the `FulfillStorageClassClaim` service on the provider server with StorageClassClaim as a request message.
+			_, err = providerClient.FulfillStorageClassClaim(
+				r.ctx,
+				r.storageCluster.Status.ExternalStorage.ConsumerID,
+				r.storageClassClaim.Name,
+				r.storageClassClaim.Spec.EncryptionMethod,
+				storageClassClaimStorageType,
+			)
+			if err != nil {
+				// Reset StorageClassClaim phase on error since we want the reconciliation loop to enter this block the next time.
+				r.storageClassClaim.Status.Phase = ""
+				return reconcile.Result{}, fmt.Errorf("failed to initiate fulfillment of StorageClassClaim: %v", err)
+			}
+		}
+		r.storageClassClaim.Status.Phase = ocsv1alpha1.StorageClassClaimConfiguring
+
+		// Configuration phase
+		// Triggered only if the corresponding StorageClass is not found.
+		// TODO Possible to check for Secret, in addition to existing StorageClass check (secret name is dynamic)?
+		correspondingStorageClassDoesNotExists := errors.IsNotFound(
+			r.get(&storagev1.StorageClass{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: r.storageClassClaim.Name,
+				},
+			}),
+		)
+		if r.storageClassClaim.Status.Phase == ocsv1alpha1.StorageClassClaimConfiguring && correspondingStorageClassDoesNotExists {
```

Comment:
you know that ` r.storageClassClaim.Status.Phase == ocsv1alpha1.StorageClassClaimConfiguring` because you just set it up like that a couple of lines above this one.

Also, `correspondingStorageClassDoesNotExists` is not a good indicator to skip this function because an external actor can modify the storageclass and it is the responsibility of this controller to revert the change. 

Let's remove this if and run this code on every reconciliation!